crypto/fipsmodule/ec/wnaf.c

```diff
@@ -85,8 +85,17 @@
 //   http://link.springer.com/chapter/10.1007%2F3-540-45537-X_13
 //   http://www.bmoeller.de/pdf/TI-01-08.multiexp.pdf
 
-void ec_compute_wNAF(const EC_GROUP *group, int8_t *out,
-                     const EC_SCALAR *scalar, size_t bits, int w) {
+// Returns the value of the bit at |idx| place in the scalar.
+static int ec_scalar_bit_at_idx(const EC_SCALAR *s, size_t s_bits, size_t idx) {
```

Comment:
What is the need for this function? Isn't it similar to the function that was used https://github.com/aws/aws-lc/blob/85f58da29e68906d094d378fe968f6b90807d05f/crypto/fipsmodule/bn/shift.c#L261 or is it to inline it?

Reply:
`ec_compute_wNAF` calls `bn_is_bit_set_words` with `group->order.N.width` which is why it needs `group` as an input argument. Since `ec_nistp_scalar_mul_public` doesn't have access to the group and I wanted to be able to reuse `ec_compute_wNAF` there, I had to remove `group` from `ec_compute_wNAF`'s arguments. So I had to add a replacement for `bn_is_bit_set_words` that works with the bit size of the scalar instead of the number of words.

I hope this makes sense :)

Reply:
I guessed this was the reason, but I thought you could call `bn_is_bit_set_words` with the second argument being `bits` as in the current call (the first argument would be `s->words`, I believe), no?

Reply:
no no, the second argument of `bn_is_bit_set_words` is number of words, not bits.

Reply:
Discussed offline. @dkostic reverted back to using `bn_is_bit_set_words` after calculating the number of words to pass to it.